### DISCUSSION TITLE
Order of async operations

### DISCUSSION THREAD

**Discussion by ghost** [coronabytes/dotnet-arangodb#30] [Q&A] [answered]
Hello
A question about order of async operations. Lets say I need to drop and then recreate a number of collections and graphs. But I'd rather not block my application on each operation. Rather I'd like to store all Task objects and then use Task.WhenAll.
From what I see in [Arango docs](https://www.arangodb.com/docs/stable/http/async-results-management.html), there's no guarantee about operations order. Given that your library works through REST API in async mode, I assume it also doesn't guarantee any order of operations.
Is it true and I have to gate operations (drops from creates) by Waits or not?

**Comment by coronabytes** [comment, marked as answer]
why'd you assume "blocking" your application is bad when it's infact async?

Task.WhenAll wont help you - it needs to happen in order (at least the drop and create phases).
maybe you want to take a look at Core.Arango.Migration package

I had batch operations in the driver - however the arango team deprecated the endpoint rather quickly after I noted some errors
